components/post_view/post_body_additional_content/post_body_additional_content.jsx

```diff
@@ -84,15 +88,27 @@ export default class PostBodyAdditionalContent extends React.PureComponent {
 
     componentDidMount() {
         // check the availability of the image rendered(if any) in the first render.
+        this.loadShortenedImageLink();
         this.preCheckImageLink();
     }
 
+    async loadShortenedImageLink() {
+        if (!this.isLinkImage(this.state.link)) {
+            const {data} = await this.props.actions.getRedirectLocation(this.state.link);
+            if (data && data.location) {
+                this.setState({link: data.location});
```

Comment:
This needs some protection on it in case either the component unmounts or the link changes. The first one just needs a `this.mounted` that you check before calling setState, but for the second, you'll probably want to save `this.state.link` and then use the function form of setState to make sure that the link hasn't changed for some other reason like
```javascript
const link = this.state.link;
const {data} = await ...;
this.setState((state) => {
    if (state.link !== link) {
        return {};
    }

    return {link: data.location};
});
```